Yank/repex.py

```diff
@@ -2101,7 +2101,7 @@ def _propagate_replica(self, replica_id):
             prefix = 'nan-error-logs/iteration{}-replica{}-state{}'.format(
                 self._iteration, replica_id, thermodynamic_state_id)
             e.serialize_error(prefix)
-            raise
+            raise utils.SimulationNaNError
```

Comment:
I think you're returning the class here without instantiating.

Reply:
You're right, the `raise` statement actually does instance the `Exception` class technically, but is not the PEP8/Pyhtonic way to do it, so I'll instance it.